VIPs/vip-10.md

```diff
@@ -31,40 +31,35 @@ The current storage node database is centralized and needs to be stored on-chain
 
 ### StorageNode
 
-- didAddress: string
-- endpointUri: string
-- countryCode: string
-- regionCode: string
-- datacenterId: id
-- lat: int
-- long: int
-
-1. `didAddress`: The DID that is associated with the storage node. Must be only one `didAddress` registered at any time.
-2. `endpointURI`: The storage node endpoint. Must be only one `endpointURI` registered at any time.
-3. `countryCode`: Unique two-character string code (see [VIP-7](./vip-7.md))
-4. `regionCode`: Unique region string code (see [VIP-7](./vip-7.md))
-5. `datacenterId`: Unique datacenter identifier. Must match an existing datacenter available via `getDatacenters()`.
-6. `lat`: Latitude value multiplied by 10^8
-7. `long`: Longitude value multiplied by 10^8
+- `didAddress` (string): The DID that is associated with the storage node. Must be only one `didAddress` registered at any time.
+- `endpointUri` (string): The storage node endpoint. Must be only one `endpointURI` registered at any time.
+- `countryCode` (string): Unique two-character string code (see [VIP-7](./vip-7.md))
+- `regionCode` (string): Unique region string code (see [VIP-7](./vip-7.md))
+- `datacenterId` (id): Unique datacenter identifier. Must match an existing datacenter available via `getDatacenters()`.
+- `lat` (int): Latitude value multiplied by 10^8
+- `long` (int): Longitude value multiplied by 10^8
+- `numberslots` (int): The number of storage slots available on this storage node
 
 ### Datacenter
 
-- id: int
-- countryCode: string
-- regionCode: string
-- lat: int
-- long: int
+- `id` (int)
+- `countryCode` (string)
+- `regionCode` (string)
+- `lat` (int)
+- `long` (int)
 
 ## State variables
 
-TBA
+- `STAKE_PER_SLOT` (const): The number of tokens that must be staked by a storage node operator for each storage slot they make available to the network. `default=60`. Only contract owner can change at this stage. _In the future this will become dynamic based on market dynamics._
 
 ## addNode(nodeInfo: StorageNode, requestSignature: string, requestProof: string, authSignature: string)
 
 Storage nodes must register themselves with the protocol so they can be discovered by new users. Storage nodes provide additional metadata (via `/status` endpoint) that assists users select the most appropriate storage nodes to use.
 
 It's not possible to re-use a DID to register multiple storage nodes. This is because storage endpoints sign their responses using the DID private key to enhance security of the protocol.
 
+The requester must provide VDA tokens to stake. The number of staked tokens must be `numberSlots` * `STAKE_PER_SLOT`. The staked tokens will be registered against the storage node being registered so they can be unstaked in the future (see `withdraw()`)
```

Comment:
The `withdraw()` function can unstake only excess token when the `STAKE_PER_SLOT` is updated.
Staked tokens will be released by `removeNodeComplete()` function.

Reply:
I thought the comments in `withdraw()` and `removeNodeComplete()` made this clear?

Reply:
How about to update the comments in the parenthesis as following?
```
(see `withdraw()`)
```
to
```
(see `withdraw` and `removeNodeComplete`)
```

Reply:
Got it. Good idea.